typings/bson/json_util.pyi

```diff
@@ -1,4 +1,5 @@
 from typing import Any
 
 def loads(s: str) -> Any: ...
-def dumps(obj: object) -> str: ...
+def dumps(obj: object, *args, **kwargs) -> str: ...
+def default(obj: object, json_options: Any = ...) -> dict: ...
```

Comment:
```suggestion
def default(obj: object, json_options: Any = ...) -> Any: ...
```

Looking at the source code for  `default`, it sometimes returns `SON`, `dict[str, Any]`, `bytes` depending on the param

we could add a bunch of `overload`s, but fine to type it as returning `Any` for now

https://github.com/mongodb/mongo-python-driver/blob/acfa7b615c5d263a2fc94d7746771beb6fff4016/bson/json_util.py#L766-L859

Reply:
Yeah, I noticed the dicts only :))